cvat-core/src/frames.js

```diff
@@ -79,87 +81,107 @@
+                                } catch (_) {
+                                    reject(this.number);
+                                }
+                            }
+                        }
+                    } else {
+                        if (this.number % chunkSize > 1){
+                            if (!provider.isNextChunkExists(this.number)){
+                                const nextChunkNumber = Math.floor(this.number / chunkSize) + 1;
+                                provider.setReadyToLoading(nextChunkNumber);                            
+                                serverProxy.frames.getData(this.tid, nextChunkNumber).then(nextChunk =>{
+                                provider.startDecode(nextChunk, (nextChunkNumber) * chunkSize, (nextChunkNumber + 1) * chunkSize - 1, function(){});
+                                });
+                            }
+                        }
+                        resolve(frame);
+                    }
+                } catch (exception) {
+                    if (exception instanceof Exception) {
+                        reject(exception);
+                    } else {
+                        reject(new Exception(exception.message));
+                    }
                 }
-            } catch (error) {
-                reject(error);
-            }
-        });
+
+            return new Promise(getFrameData.bind(this));
+        }
+
+       
     }
+};
 
-    async function getFrame(taskID, mode, frame) {
+    async function getFrame(taskID, chunkSize, mode, frame) {
         if (!(taskID in frameDataCache)) {
-            frameDataCache[taskID] = {};
-            frameDataCache[taskID].meta = await serverProxy.frames.getMeta(taskID);
+            const blockType = mode === 'interpolation' ? cvatData.BlockType.TSVIDEO
+                : cvatData.BlockType.ARCHIVE;
+
+            const value = {
+                meta: await serverProxy.frames.getMeta(taskID),
+                chunkSize,
+                provider: new cvatData.FrameProvider(3, blockType, chunkSize),
+            };
 
             frameCache[taskID] = {};
+            frameDataCache[taskID] = value;
         }
 
-        if (!(frame in frameDataCache[taskID])) {
-            let size = null;
-            if (mode === 'interpolation') {
-                [size] = frameDataCache[taskID].meta;
-            } else if (mode === 'annotation') {
-                if (frame >= frameDataCache[taskID].meta.length) {
-                    throw new ArgumentError(
-                        `Meta information about frame ${frame} can't be received from the server`,
-                    );
-                } else {
-                    size = frameDataCache[taskID].meta[frame];
-                }
-            } else {
-                throw new ArgumentError(
-                    `Invalid mode is specified ${mode}`,
-                );
-            }
+        let size = null;
+        [size] = frameDataCache[taskID].meta;
```

Comment:
This is true only if mode === interpolation (you have removed this condition)

Reply:
In case of annotation mode size = frameDataCache[taskID].meta[frame];
And also we have to check 
```
if (frame >= frameDataCache[taskID].meta.length) {
                    throw new ArgumentError(
                        `Meta information about frame ${frame} can't be received from the server`,
                    );
                }
```
You also have deleted it

Reply:
And you have deleted the check:
```
                throw new ArgumentError(
                    `Invalid mode is specified ${mode}`,
                );
```